code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -92,6 +92,19 @@
 /datum/action/xeno_action/plant_weeds/slow
 	cooldown_timer = 12 SECONDS
 
+/datum/action/xeno_action/plant_weeds/slow/action_activate()
+	var/mob/living/carbon/xenomorph/hivemind/hiveminde = owner
+	if(hiveminde)
+		if(!locate(/obj/effect/alien/weeds) in range(1, hiveminde.loc))
+			var/obj/effect/alien/weeds/nearbies = locate() in range(7, hiveminde.loc)
+			if(nearbies)
+				hiveminde.forceMove(get_turf(nearbies))
+			else
+				hiveminde.forceMove(get_turf(hiveminde.core))
+			to_chat(hiveminde, "<span class='xenonotice'>you can't place weeds with no weeds nearby, you got moved to [nearbies ? "the closest weed" : "your core"].</span>")
+			return
+ 	return ..()
```

Comment:
```suggestion
	var/mob/living/carbon/xenomorph/hivemind/hiveminde = owner
	if(!locate(/obj/effect/alien/weeds) in range(1, hiveminde.loc))
		var/obj/effect/alien/weeds/nearbies = locate() in range(7, hiveminde.loc)
		if(nearbies)
			hiveminde.forceMove(get_turf(nearbies))
		else
			hiveminde.forceMove(get_turf(hiveminde.core))
		to_chat(hiveminde, "<span class='xenonotice'>you can't place weeds with no weeds nearby, you got moved to [nearbies ? "the closest weed" : "your core"].</span>")
			return
 	return ..()
```
?

Reply:
huh ?

Reply:
if(hiveminde) to make sure its a hiveminde if someone ever wants to add that slow ability to anything else(shoulnd't) and avoid runtimes by not having hiveminde

Reply:
A) Hiveminde is not istyped
B) something like this shouldnt be istyping in the first place

Reply:
it doens't need istyping because it already is the type, but i can do that ye

Reply:
Owner is always set why are you doing this

Reply:
for hivemind.core

Reply:
im aware but if(hiveminde) will always be true so theres no point to having it

Reply:
ok fair